gen_code_can_common.c

```diff
@@ -0,0 +1,820 @@
+                           uint8_t quality, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = num_sats; 
+    output->data[4] = quality; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_fill_lvl_msg(uint32_t timestamp,
+                                                   uint8_t level, 
+       enum FILL_DIRECTION fill_direction, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = level; 
+    output->data[4] = (uint8_t) fill_direction; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+bool build_radi_value_msg(uint32_t timestamp,
+                                                   uint8_t radi_board, 
+                           uint16_t radi, 
+
+                        can_msg_t *output)
+
+{
+    if (!output) { return false; }
+
+    output->sid = MSG_GPS_ALTITUDE | BOARD_UNIQUE_ID;
+    write_timestamp_3bytes(timestamp, output);        
+    
+        output->data[3] = radi_board; 
+    output->data[4] = radi; 
+
+    
+    output->data_len = 5;
+
+    return true;
+}
+        
+int get_general_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+int get_actuator_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_alt_arm_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    if (!altimeter) { return false; } 
+
+
+}
+
+int get_reset_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_debug_msg(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    if (!level) { return false; } 
+    if (!line) { return false; } 
+
+
+}
+
+bool get_debug_printf(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_debug_radio_cmd(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+int get_actuator_status(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_alt_arm_status(const can_msg_t *msg,
+                                                   uint16_t drogue_v, 
+                           uint16_t main_v, 
+)
+
+{
+    if (!msg){ return false; }
+    if (!altimeter) { return false; } 
+    if (!drogue_v) { return false; } 
+    if (!main_v) { return false; } 
+
+
+}
+
+int get_general_board_status(const can_msg_t *msg,
+                        )
+
+{
+    if (!msg){ return false; }
+    
+
+}
+
+bool get_sensor_temp(const can_msg_t *msg,
```

Comment:
name: get_temp_data
params: `sensor_id` instead of `*sensor_num`, `uint24_t` instead of `int32 *temp`
missing:
`*sensor_num = msg->data[3];
    //handle 24 bit to 32 bit sign conversion
    *temp = (msg->data[4] & 0x80)? 0xFF << 24 : 0;
    *temp |= ((uint32_t)msg->data[4] << 16);
    *temp |= ((uint32_t)msg->data[5] << 8);
    *temp |= msg->data[6];

    return true;
}`